node/src/components/consensus/highway_core/highway.rs

```diff
@@ -574,6 +572,19 @@ impl<C: Context> Highway<C> {
         evidence_effects.extend(self.on_new_unit(&unit_hash, now, rng));
         evidence_effects
     }
+
+    /// Adds endorsements to the state.
+    fn add_endorsements(&mut self, endorsements: Endorsements<C>) -> Vec<Effect<C>> {
+        // TODO: Not faulty — evidence!
```

Comment:
Hm? I don't understand.

Reply:
Sorry, I forgot about that `TODO`! What I meant was: We're not interested in who's faulty but in whom we have direct evidence against. If e.g. we only had _indirect_ evidence (from an older era), and due to these endorsements we now have _direct_ evidence, then we did create a new vertex and want to gossip that.
But I'll just try what you suggested below anyway instead.